capacity4more/modules/c4m/widgets/c4m_document_widget/c4m_document_widget.module

```diff
@@ -157,6 +157,31 @@ function c4m_document_widget_entityreference_autocomplete_tags_validate($element
   }
   // Update the value of this element so the field can validate the product IDs.
   form_set_value($element, $value, $form_state);
+
+
+  $field_values = array();
+  preg_match_all("/\((\d+)\)/", $element['#value'], $field_values);
+
+  $settings['c4m']['data']['relatedDocuments'] = $field_values[1];
+
+  // Unset previous values.
+  foreach ($form['#attached']['js'] as $id => $data) {
+    if (isset($data['data']['c4m'])) {
```

Comment:
instead of this scary IFs you can just have a single ``!empty($data['data']['c4m']['data']['relatedDocuments']['relatedDocuments'))``

Also explain in a comment why you do this (because the form validation, etc')